sources/Platform/Linux/LinuxDisplay.cpp

```diff
@@ -134,187 +88,73 @@ bool Display::IsCursorShown()
 
 bool Display::SetCursorPosition(const Offset2D& position)
 {
-    LinuxSharedX11DisplaySPtr sharedX11Display = LinuxSharedX11Display::GetShared();
-    ::Display* dpy = sharedX11Display->GetNative();
-    ::Window rootWnd = DefaultRootWindow(dpy);
-    XWarpPointer(
-        /*display*/     dpy,
-        /*src_w*/       None,
-        /*dest_w*/      rootWnd,
-        /*src_x*/       0,
-        /*src_y*/       0,
-        /*src_width*/   0,
-        /*src_height*/  0,
-        /*dest_x*/      position.x,
-        /*dest_y*/      position.y
-    );
-    XFlush(dpy);
-    return true;
+    return g_primaryDisplay->SetCursorPositionInternal(position);
 }
 
 Offset2D Display::GetCursorPosition()
 {
-    LinuxSharedX11DisplaySPtr sharedX11Display = LinuxSharedX11Display::GetShared();
-    ::Display* dpy = sharedX11Display->GetNative();
-    ::Window rootWnd = DefaultRootWindow(dpy);
-    ::Window rootWndReturn, childWndReturn;
-    unsigned int mask;
-    Offset2D rootPosition = { 0, 0 };
-    Offset2D childPosition = { 0, 0 };
-    XQueryPointer(
-        /*display*/         dpy,
-        /*w*/               rootWnd,
-        /*root_return*/     &rootWndReturn,
-        /*child_return*/    &childWndReturn,
-        /*root_x_return*/   &rootPosition.x,
-        /*root_y_return*/   &rootPosition.y,
-        /*win_x_return*/    &childPosition.x,
-        /*win_y_return*/    &childPosition.y,
-        /*mask_return*/     &mask
-    );
-    return rootPosition;
+    return g_primaryDisplay->GetCursorPosition();
```

Comment:
Oh I think this needs to be `GetCursorPositionInternal()`, otherwise this would be an infinite recursion.